%!TEX root = ../dissertation.tex
\begin{savequote}[60mm]
There is no scientific study more vital to man than the study of his own brain. Our entire view of the universe depends on it.\qauthor{Francis Crick}
\end{savequote}

% If the human brain were so simple that we could understand it, we would be so simple that we couldn’t.
% – Emerson M. Pugh

% The human brain, then, is the most complicated organization of matter that we know.     
% Isaac Asimov

% One of the difficulties in understanding the brain is that it is like nothing so much as a lump of porridge.
% Richard L. Gregory

% The brain is a monstrous, beautiful mess. Its billions of nerve cells - called neurons - lie in a tangled web that displays cognitive powers far exceeding any of the silicon machines we have built to mimic it.
% William Allman

\chapter{microRNA Dynamics in Cholinergic Differentiation of Human Neuronal Cells}

\newthought{Even though much has been achieved} in the integrative study of \ac{mir} control of gene expression, computational analysis of transcriptional interactions has not yet reached the level of sophistication needed for the accurate prediction of events inside mammalian cells. For this reason, combination of a bioinformatical assay with modern molecular biology methods can strengthen the message and reproducibility of any approach. The spectrum of processes worthy of study is as wide as modern biomedicine. Similarly, experimental models can span the entire repertoire available to a modern laboratory. The selection of a model adequate to the research question therefore is as important as diligent analysis and careful interpretation of results.

This chapter will discuss the current state of knowledge on brain transciptomics, generally and in the specific case of cholinergic neurons in the \ac{cns}, and then go on to explain the steps I undertook to elucidate small RNA processes in central cholinergic systems. First, my aim was to clarify co-expression patterns of central cholinergic neurons, which required analysis of transcriptome data in single-cell resolution. Based on this information, I selected two human models of cholinergic neuronal differentiation and established a differentiation protocol amenable to RNA extraction and successive molecular biology assays, most importantly, \ac{seq}. The expression patterns so obtained were then used to perform bioinformatics analyses using the database introduced in Chapter \ref{sec:database:mirnet}, \textit{miRNet}. 

\section{Neuronal Transcriptomes - Background}
The mammalian brain requires a constant supply of oxygen and nutrients, because it does not provide storage for either. Though it only makes up approximately 2\% of the entire human body mass, its energy expenditure is around 20\% of the whole\cite{Raichle2002}. For this reason, each square millimetre of brain tissue (except for the ventricles) is infiltrated by hundreds of capillaries\cite{Bohn2016}. Since the blood-brain-barrier is essentially provided by supporting glia cells surrounding all capillaries from the »inside« (see Fig. \ref{fig:bbb}, adapted from my second publication\cite{Lobentanzer2019b}), neurons constitute only a minority of brain tissues (but burn two thirds of its energy).

\begin{figure}
\includegraphics[width=\textwidth]{figures/bbb}
\caption[The blood-brain-barrier.]{\textbf{Schematic display of the blood-brain-barrier.} The blood-brain-barrier surrounds virtually every capillary in the CNS. A: Astrocyte, B: Basal Membrane, E: Endothelial Cell, T: Tight Junction. Modified from Lobentanzer \& Klein, 2019\cite{Lobentanzer2019b}.
\label{fig:bbb}}
\end{figure}

Before the advent of single-cell \ac{seq}, studies that endeavoured to clarify the transcriptional profiles of neurons used microarray technology, which was recently succeeded by \ac{seq} (also known as deep sequencing or next generation sequencing). For these methods, several cubic millimetres of brain tissue are required at the least; often, cubic centimetres are used. In contrast, the diameter of neuronal somata is usually in the micrometre range. Thus, the resolution of the method and the actual cellular resolution differ by a factor of approximately \num{1000}. Additionally, even among the neuronal population, there is considerable heterogeneity and transcriptomic plurality; single brain regions rarely consist of less than 30 different neuron types, tightly packed next to each other, each with their own transcriptional identity\cite{Darmanis2015, Zeisel2015, Tasic2016, Habib2016}. Newest studies, deciphering the murine nervous system by sequencing of \num{500000} individual cells, show that neuron diversity is very similar regardless of brain region\cite{Zeisel2018}. These circumstances hold true for any mammal, and most of our knowledge stems from the analysis of our favourite research animal, the mouse. In humans, the complexity is only exacerbated; in fact, the elevation in \ac{cns} transcriptional complexity might be the reason for our superior cognitive abilities(cite).

Cholinergic neurons always constitute a minority in any neuronal population, sometimes to extremes. Most tissues are dominated by few neuron types, such as pyramidal cells in the cortex; the most common neurotransmitter types are GABAergic (inhibitory) and glutamatergic (excitatory), each with several subtypes. It is estimated that more than 80\% of cortical neurons are excitatory, and more than 90\% of synapses release glutamate\cite{Raichle2002}. There are two major cholinergic regions in the mammalian brain: The striatum is fairly well-populated with rather large cholinergic interneurons, and the basal forebrain holds a large amount of (smaller) cholinergic projection neurons (compare Fig. \ref{fig:projections}). However, in transcriptomic analyses, these tissues are seldom used, for lack of scientific interest, or because they are notoriously hard to access (the basal forebrain is small and deeply imbedded in the midbrain). The cortex, particularly the neocortex, is most often the tissue of choice in these studies, due to its scientific interest and accessibility. Though it contains only a minuscule amount of cholinergic interneurons whose identity still is a matter of debate, several of the recent single-cell \ac{seq} approaches have independently identified cholinergic interneurons in cortical regions (see Fig. \ref{fig:singlecell}). 

All of the above taken into consideration, several limitations apply when it comes to the selection of a cellular model for the cholinergic processes we aim to understand. A multicellular model is prohibited by the novelty of the subject; possibilities include \textit{in vivo} or \textit{ex vivo} experimentation on rodents or human (3D-)cell culture with multiple cell types. While the former certainly is closest to reality, our diseases of interest display a noticeable lack of transferability from lower mammals to human(cite). The latter, on the other hand, introduces a complexity not well suited to the level of knowledge we possess about the studied processes; in addition, these models are very new, and thus, too many variables would be unknown. For these reasons, I selected two mono-cultures of human neuronal cells for my experiments. I chose to introduce a second cellular model during the experimental phase, because some of the studied diseases display a clear sexual dimorphism; the first experiments were performed on the female-originated LA-N-2 cell line, and later, to explore sex-related differences, the male-originated LA-N-5 cells were added.

\section{Cortical Single-Cell RNA Sequencing}
To estimate the potential impact of the transcriptomes studied in our model on the diseases of interest, co-expression of the relevant genes has to be asserted. Similarly, if neurokines are to possess any relevance for cholinergic properties of central nervous cells, the cells in question would have to express molecular machinery required to receive neurokine signals. The advent of single-cell \ac{seq} for the first time enables the resolution of gene expression on a cellular basis, and thus the disentangling of spatially close individual neuron types (and other, non-neuronal \ac{cns} cells); most of this information would be lost in \ac{seq} performed on brain homogenate, even of a small biopsy. Differences in genes would be reduced to the universally expressed »housekeeping« genes, and in \acp{mir}, the situation would be worse, in parallel to their even more tissue-specific expression.

For this reason, I analysed all publicly available single-cell \ac{seq} datasets relevant to our questions, which provide a detailed tally of transcriptional subtypes in the \ac{cns}. More specifically, all studies that were available at the time focused on some subsection of the cortex (visual or somatosensory) or the hippocampus. Additionally, the data provided by those studies was in some cases pre-aggregated to represent groups of single neurons with similar transcriptomes (Fig. \ref{fig:singlecell} A\&B\cite{Zeisel2015, Tasic2016}); in other cases, every single neuron was represented (Fig. \ref{fig:singlecell} C\&D\cite{Darmanis2015, Habib2016}). 

An important quality-related parameter of a single-cell \ac{seq} experiment is the sequencing depth achieved per individual sequenced cell. Some datasets I screened do not provide sufficient depth to resolve genes with medium expression, which includes our primary cholinergic markers \textit{\ac{chat}} and \textit{\ac{slc}}. The datasets which did provide adequate sequencing depth were filtered for their expression of these markers, and additionally characterised by their expression of common markers for cell types to be expected in the \ac{cns}. This resulted in a collection of transcriptomes for potentially cholinergic cells in the sampled brain regions, and allowed an assessment of the functional type and gene co-expression patterns in central cholinergic cells (Fig. \ref{fig:singlecell}).

\begin{figure}
\centering
\includegraphics[height=10cm]{figures/singlecell}
\caption[Short figure name.]{This is a figure that floats inline and here is its caption.
\label{fig:singlecell}}
\end{figure}

Most cells identified as cholinergic by this definition expressed the general neuronal marker \textit{\acs{rbf}}, also known by its trivial name NeuN, but not the microglial marker \textit{\acs{aif}}. Few cells (or clusters of cells) expressed non-neuronal markers such as \textit{\acs{gfap}} (astrocytes) or \textit{\acs{oli}} (oligodendrocytes), hinting at sparse non-neuronal cholinergic functions. In agreement with my findings, cells or clusters identified as cholinergic by the authors of the respective studies\cite{Zeisel2015, Tasic2016} (also by personal communication) had been classified as interneurons and co-expressed a number of known phenotypic neuronal markers, such as \textit{\ac{sst}} and \textit{\ac{vip}}.

The identified cholinergic cells also revealed a constant co-expression with neurokine-related genes, particularly the transmembrane neurokine receptors \ac{lifr} and \ac{ilst}, demonstrating a capacity to receive and process neurokine signals. In contrast, the high affinity receptor for \ac{ngf}, \textit{NTRK1}, is not co-expressed in mature (NeuN-positive) cholinergic neurons, fundamentally distinguishing these cells from the basal forebrain cholinergic projection neurons.
\todo{Clusters?}\todo{Permutation targeting analyses?}

\section{The Cellular Model}
\subsection{Requirements of a Cholinergic Neuron Cellular Model} \label{sec:cellculture:model}
Describe here the preliminary computational analyses leading to discovery of the CHAT miRNA anomaly, and thus the search for a CHAT-centric cell model? \todo{where goes the preliminary analysis?}
\subsection{The SH-SY5Y Neuroblastoma Cell Line}
A prominent example of human neuronal cell culture used in the identification and elucidation of cholinergic processes is the immortalised neuroblastoma cell line SH-SY5Y\cite{Biedler1978}. Derived from its parent line SK-N-SH, an adrenergic neuroblastoma\cite{Biedler1973}, it expresses ample amounts of \textit{\ac{ache}}, and thus had become a work horse in many cholinergic fields, such as Alzheimer's Disease (which is treated with \ac{ache} antagonists), pesticide development, and warfare(cite). However, in spite of its usefulness for processes involving \textit{\ac{ache}}, it turned out a less than optimal choice for the study of molecular events surrounding \textit{\ac{chat}} and \textit{\ac{slc}}, as it barely expresses both genes(cite), and cannot be coerced to elevate \textit{\ac{chat}} expression by the usual differentiation techniques (own experimentation, data not shown). Thus, for the questions asked in this chapter of the dissertation, SH-SY5Y does not qualify as adequate representation of a »cholinergic neuron«.

\subsection{The LA-N Neuroblastoma Cell Lines}
Following the elimination of SH-SY5Y as a suitable subject, I scoured the literature for candidates representing a cholinergic neuronal transcriptome, and found, among others, representatives of the LA-N neuroblastoma cell lines developed by R.C. Seeger around 1980\cite{Seeger1977, Seeger1982}. Neuroblastoma is a form of neuronal cancer often affecting small children, and, consequentially, the two cell lines used in my experiments are immortalised biopsies of a 3 year old girl (\acs{la2}\cite{Seeger1977}) and of a 4 month old boy (\mbox{\acs{la5}}\cite{Seeger1982}). The decision to use \ac{la2} as my initial cellular model was influenced by three factors: it is well described in literature, although most studies had been published in the 1980s and 90s; it expresses a substantial amount of \textit{\ac{chat}} and \textit{\ac{slc}}; and it responds to neurokine-mediated differentiation by assuming a neuronal morphology accompanied by further elevation of \textit{\ac{chat}} and \textit{\ac{slc}} expression. \ac{la5} was not nearly as well described as \ac{la2}, but later added to the experimental roster because of the complementary sex and hints towards cholinergic differentiation under retinoic acid\cite{Hill1997}.

\subsection{Culture}
LA-N cell culture is not as straightforward as many »go-to« human cell lines used in today's laboratories. Because \ac{la2} and \ac{la5} are very similar in this regard, they were treated similarly and I will describe the procedure for both. They have comparatively high duplication times, which can be lowered by using certain conditions that affect medium composition, nutrition, and CO\textsubscript{2} content. The cells were acquired at DSMZ (Braunschweig, Germany), which recommends keeping them in a 50:50 mixture of \ac{dmem} and \ac{rpmi}, with 20\% \ac{fcs} added. Sometimes, recommendations also suggest Leibovitz's L-15 medium, which is specifically designed for low CO\textsubscript{2} conditions, and others have suggested increased CO\textsubscript{2} levels inside the incubator. I found a combination of the DSMZ-recommended medium with 8\% CO\textsubscript{2} atmosphere inside a 37°C incubator to accelerate growth to a degree that the cells could be split 1:3 to 1:4 in a weekly cycle. This protocol was used for all further experiments, which were performed between splits 2 to 8 after thawing of a batch from -80°C. All handling during maintenance and experimentation was performed under a laminar flow hood.

\subsection{Differentiation}
Neuronal differentiation of neuroblastoma cell lines has been performed in many instances, utilising a wide variety of differentiation agents such as the very general retinoic acid or 5-bromo-uracil, or very specific reagents, such as the neurokines \ac{il} and \ac{cntf}(cite). LA-N cells have also been described to react to a selection of these substances; however, due to the elevated interest in neurokine mechanisms, I chose to go ahead with a reagent from that group. Somewhat arbitrarily, I used \ac{cntf}, because upon my inquiry, James McManaman revealed in personal communication that the »\textit{\ac{chat}} development factor« that he had discovered\cite{McManaman1988} was, in fact, \ac{cntf}, which had never been published. Additionally, of the neurokines used for differentiation purposes, \ac{cntf} is best described in literature and easily acquired in dried form from Merck (formerly SigmaAldrich, Darmstadt, Germany). \ac{cntf} was resuspended in pure water to a concentration of \SI{25}{\micro\gram\per\milli\litre} and stored for experimentation in aliquots at -20°C.

LA-N cells are very sensitive to repeated temperature changes (or other handling-related disturbances), which resulted in increased amounts of apoptotic cells following repeated removal from the incubator after seeding or medium changes during the experiment. For this reason, I chose to only add the differentiation reagent once, 24h after initial seeding of the cells into the wells of a 12-well-plate, and avoid further disturbances until the time of lysis. For the maximum duration of my experiments, 120h from seeding until lysis, the initially supplied medium was sufficient for survival.

Differentiation was performed in regular growth medium without changes in \ac{fcs} content, and \ac{cntf} was added to the medium after an initial growth period of 24h. Cells were seeded into 12-well plates at approximately \num{200000} cells/well, with 1 ml of growth medium. To determine the optimal amount of \ac{cntf} for differentiation, time-dose curves were determined for both cell lines in a range from \SIrange{1}{100}{\nano\gram\per\milli\litre}. Here, I discovered the first pharmacological difference between \ac{la2} and \ac{la5}: the maximum of their cholinergic response to neurokine stimulation (i.e., an elevation in \textit{\ac{chat}} and \textit{\ac{slc}} transcription) occurs at different concentrations of \ac{cntf}. While \ac{la2} cells respond most strongly to \SI{100}{\nano\gram\per\milli\litre}, \ac{la5} cells show an »inverted u«-type dose response with a maximum around \SI{10}{\nano\gram\per\milli\litre} \ac{cntf} (Fig. \ref{fig:time-dose}). James McManaman, who studied LA-N differentiation thoroughly in the 1990s\cite{McManaman1991}, believes both lines to respond in an »inverted u«-type manner (personal communication); thus, I assume that the \ac{la2} response would also diminish at \ac{cntf} concentrations significantly higher than \SI{100}{\nano\gram\per\milli\litre}. I also suspect that CNTF concentrations could be significantly lowered by removal of the high amount of \ac{fcs} in the medium, however, that would likely require the use of a special serum-free medium, which would have to be established up front, and might have other, unforeseen consequences. Regardless, \ac{cntf} concentrations around \SI{100}{\nano\gram\per\milli\litre} (i.e., pico- to nano-molar) still are well within the physiological range of concentrations that the mammalian brain is able to reach by paracrine secretion via, e.g., astrocytes\cite{Sun2016}.

\begin{figure}
\includegraphics[width=\textwidth]{figures/time-dose}
\caption[Time-dose curve, LA-N-2 and LA-N-5.]{\textbf{Time-dose curve of CNTF-mediated differentiation of LA-N-2 and LA-N-5.}
\label{fig:time-dose}}
\end{figure}

To study the small RNA dynamics following \ac{cntf} exposure of \ac{la2} and \ac{la5}, I selected 4 time points at which the experiment was stopped and the cells were quickly lysed \textit{in situ} to preserve total RNA in that state: for the quick, immediate-early-like phase, at 30 and 60 minutes after the addition of \ac{cntf}, and, for the long-term effects of differentiation, at 48 and 96 hours after the addition of \ac{cntf} (Fig. \ref{fig:timepoints}, from my first publication\cite{Lobentanzer2019a}). Each time point was controlled by a pseudo-treated (using pure water) culture from the same batch that had been seeded at the same time as the experimental group. In the final series used for the parallel sequencing of \ac{la2} and \ac{la5}, all experiments were carried out in quadruplicates. 

\begin{figure}
\centering
\includegraphics[width=0.8\textwidth]{figures/timepoints}
\caption[LA-N-2 / LA-N-5 Differentiation Timeline.]{\textbf{LA-N-2 / LA-N-5 Differentiation Timeline.}
\label{fig:timepoints}}
\end{figure}
\todo{add early/late}

\subsection{RNA Isolation}
Total RNA was isolated using TRIzol (ThermoFisher Scientific), essentially as suggested by the manufacturer, with slight changes to the protocol to enrich small RNA species. The cells, growing in a monolayer in 12-well-plates, were cleared of medium, washed two times with \SI{500}{\micro\litre} of cell culture grade \ac{pbs} (Gibco), and immediately suspended in \SI{1}{\milli\litre} of TRIzol, pipetting up and down until visibly dissolved. After incubation for 5 minutes at room temperature, the samples were stored in -20°C for short periods of time until RNA isolation.

TRIzol-suspended lysates (\SI{1}{\milli\litre}) were added to RNA-separation centrifuge tubes (PhaseMaker Tubes, ThermoFisher Scientific), adding \SI{200}{\micro\litre} of pure chloroform and mixing vigorously for 15 seconds. After two minutes, the mixture was centrifuged at \SI{12000}{\g} and 4°C for 15 minutes, and the upper, watery phase containing the RNA was extracted. This was mixed with approximately 2 parts of pure ethanol and incubated for 10 minutes at room temperature to precipitate the RNA. The precipitate was spun at \SI{12000}{\g} and 4°C for another 10 minutes, and the supernatant discarded. The pellet was washed with 85\% ethanol (vortexed briefly) and centrifuged again for 5 minutes at \SI{7500}{\g} and 4°C.

After the final centrifugation step, the samples were transferred to the laminar flow hood, and air dried after removal of most of the supernatant via micropipettors. The pellet was allowed to dry almost until completion and resuspended in \SIrange{30}{50}{\micro\litre} pure RNase-free water. RNA concentration was measured at a Nanodrop 2000 instrument (ThermoFisher Scientific) and samples were diluted to a uniform concentration of \SI{100}{\nano\gram\per\micro\litre}. Finally, RNA samples were aliquoted according to later purpose and stored at -80°C.

RNA quality was determined by analysis on a 2100 Bioanalyzer instrument (Agilent) using a nano chip and \SI{1}{\micro\litre} of sample; \ac{rin} was near optimal for all samples ($> 9$).

\section{Small RNA Sequencing and Differential Expression Analysis}
\newthought{For the detection and analysis of small RNA species}, \ac{seq} is the current gold standard method. It allows the mapping of a comprehensive transcriptome and thus is vastly superior to small scale and consecutive methods such as \ac{pcr}, and even the larger scale microarrays. Microarrays, while also potentially allowing a "snapshot" of entire transcriptomes, are limited by the predetermined sequences on the chip. \ac{seq}, on the other hand, is not biased towards any structural property of the sample; this is particularly important in the analysis of small RNA species, since their sequences are very variable (\acp{trf}) and still not completely catalogued (\acp{mir}). Assuming an adequate sequencing depth ($\gtrsim 1$ M reads/sample), \ac{seq} allows a comparison of all expressed small RNA species at once, which is immensely helpful when dealing with processes on the combinatorial scale of \ac{mir} regulation.

\subsection{Sequencing}
For small RNA sequencing, the aliquoted samples were shipped on dry ice to the cooperating institute at the Hebrew University of Jerusalem, the Silberman Institute of Molecular Biology, the laboratory of Prof. Hermona Soreq. \SI{600}{\nano\gram} of total RNA per sample were prepared for sequencing using the NEBNext Small RNA Library Prep Set for Illumina (New England BioLabs). The libraries were multiplexed with coloured barcodes, allowing for sequencing of all 48 samples on one chip. Briefly, this includes ligation of sequencing adapters to both 3' and 5' ends of all (single-stranded) RNA fragments in the sample, followed by 12-15 cycles of reverse transcription to form the RNA library. Ligated and amplified libraries were then size selected via gel electrophoresis on a 6\% Polyacrylamide gel. The band representing small RNA species on the gel was excised and prepared for loading onto the sequencing chip. After loading, the chip was sequenced in a NextSeq 550 series instrument (Illumina) with a read length of 80 bases, single-ended.

Sequencing quality was determined by analysis of the raw reads using the fastqc software(cite). \todo{describe quality parameters} \todo{examples?}

\subsection{Sequence Alignment}
Raw reads were adapter-trimmed and quality filtered using the flexbar software\cite{Roehr2017} with parameters

\begin{center}\texttt{-a adapters.fa -q TAIL -qf sanger -qw 4 \\-min-read-length 16 -n 1 --zip-output GZ}\end{center} 

The sequence used in the \textit{adapters.fa} file, as recommended by the manufacturer, was $$AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC$$ 

For the alignment of \ac{mir} sequences, parts of the miRExpress 2.0\cite{Wang2009} pipeline were used according to the documentation. First, a lookup table for the current miRBase version 21 was created as per the instructions of the authors. The alignment was then performed using the commands \textit{Raw\_data\_parse}, \textit{statistics\_reads}, \textit{alignmentSIMD}, and \textit{analysis}; \textit{Trim\_adaptor} \todo{describe what miRExpress does?} was skipped because the adapters had already been trimmed in the quality filtering step. Additionally, since miRExpress is not accepting of sequences of any length, the raw data was length filtered to include only reads up to a length of 25 bases before input into miRExpress. Thus, raw reads were aligned to the miRnome provided by miRBase v21, yielding count tables of mature \acp{mir} and miRNA precursors for each sample. In total, 1913 mature \acp{mir} from miRBase v21 were discovered in the data.

\subsection{Differential Expression Analysis - R/DESeq2} \label{sec:cellculture:deseq}
To determine the effect and dynamics of \ac{cntf}-mediated differentiation of \ac{la2} and \ac{la5}, the expression state of each measured time point was compared to the respective control using the established R package DESeq2\cite{Love2014}. DESeq2 determines differential expression in count-based data by application of a linear regression model to a negative binomial distribution based on a fitted mean $\mu$ and a gene-specific dispersion value $\alpha$ (for gene $i$ and sample $j$). The mean is derived using a sample-specific »size factor«, $s$, and a parameter $q$ proportional to the expected true concentration of RNA fragments in the sample. The DESeq2 differential expression pipeline is composed of the following commands:
\begin{itemize}
\item \texttt{estimateSizeFactors()} (to estimate $s_j$)
\item \texttt{estimateDispersion()} (to estimate $\alpha_i$)
\item \texttt{nbinomWaldTest()} (application of a generalised linear model to determine log-fold changes and statistics via the Wald test, using $\mu_{ij} = s_jq_{ij}$ and $log_2(q_{ij}) = x_j\beta_i$)
\end{itemize}
The Wald test, named after Abraham Wald, is an approach to hypothesis testing that measures the distance between the tested unrestricted estimate and the null hypothesis, using the precision as a weighting factor. The larger the distance between tested values and the null, the more likely the measured values are »true«. \ac{seq} data can be modelled using binomial distributions\cite{Bullard2010}, such as the Poisson distribution, and the difference between two Poisson means (e.g., »treated« vs »control«) can be tested by generalised linear models based on the distributions themselves, Fisher's exact test, or the likelihood ratio test. However, comparative analysis has shown that the Wald test on log-transformed data provides statistical power superior to these other methods\cite{Chen2011}, particularly in lowly expressed fragments. The design formula for the linear regression was a simple combination of condition and time point and applied to \ac{la2} and \ac{la5} separately: $$y \sim condition\_time$$

To reduce the noise introduced by the high variance in low-count genes while preserving large, »real« differences, the authors propose the »shrinkage« of log-fold changes to avoid arbitrary low-cut filtering at a predefined expression (count) value. Multiple variants are available; for \ac{mir} data, the adaptive algorithm »apeglm«\cite{Zhu2019} (adaptive t prior shrinkage estimator) yielded sensible results (see Fig. \ref{fig:apeglm-comp-la2d4}).

\begin{figure}
\centering
\includegraphics[width=\textwidth]{figures/apeglm-comp-la2d4}
\caption[MD Plot Shrinkage Comparison.]{\textbf{MD Plot Shrinkage Comparison.} A mean-difference plot (MD Plot) is a plot of log-intensity ratios (differences, »M-values«) versus log-intensity averages (means, »A-values«); it is synonymous with »MA Plot«. The DESeq2 function plotMD shows the fold changes attributable to a given variable over the mean of normalised counts for all the samples in the data set. Points will be coloured red if the adjusted p value is less than 0.1. Points which fall out of the window are plotted as open triangles pointing either up or down. The left plot is generated from the standard linear model, the plot on the right is corrected by the »apeglm« algorithm\cite{Zhu2019} to reduce noise in the low-count fragments.
\label{fig:apeglm-comp-la2d4}}
\end{figure}

\subsection{microRNA Dynamics in CNTF-mediated Cholinergic Differentiation of LA-N-2 and LA-N-5}
Differential expression analysis performed in this manner yielded 490 \ac{de} \acp{mir} across all groups, with characteristic distributions between cell lines and time points. The raw data and processed counts were deposited to \ac{ncbi} \ac{geo}, accession GSE132951. An earlier sequencing experiment (deposited as GSE120520), which was similar in principle, but only comprised three biological replicates and only \ac{la2}, reproduced 80\% of \ac{de} \acp{mir} in the newer experiment. Considering the general reproducibility of \ac{seq} and the lower replicate number, 80\% is an excellent substantiation of the result.

\subsubsection{Differential Expression in Both Cell Lines}

%\begin{wrapfigure}{r}{0.5\textwidth}
%  \begin{center}
%    \includegraphics[width=0.48\textwidth]{birds}
%  \end{center}
%  \caption{Birds}
%\end{wrapfigure}

114 mature \acp{mir} were detected as \ac{de} in both cell lines, with some changes similar in both, while others were inverted (Fig. \ref{fig:countchangecor}). In both cases, however, count-change values (see Box 1) correlated highly between the two cell lines (similar: 76 \acp{mir}, Spearman’s $\rho = 0.9066$, p < 2.2E-16; inverted: 38 \acp{mir}, $\rho = 0.9294$, p < 2.2E-16).

\begin{figure}
\centering
\includegraphics[width=0.5\textwidth]{figures/countchangecor}
\caption[LA-N-2 / LA-N-5 Count-change Correlation.]{\textbf{LA-N-2 / LA-N-5 Count-change Correlation.}
\label{fig:countchangecor}}
\end{figure}

\begin{shaded}
\textbf{Box 1: The count-change metric.}

The frequently used log-fold change metric is not ideally suited for assessing the potential effect of expression changes for individual \acp{mir} because it does not reflect mean expression levels. To determine the absolute change in expression, I introduced the count-change metric, a combination of base mean expression and log-fold change, to weigh DE miRNAs against one another. The count-change is defined as follows: $$CC = (BM \cdot 2^{LFC}) - BM$$
CC: count-change, BM: base mean expression, LFC: log-2-fold-change.
\end{shaded}

\subsubsection{Differential Expression Along The Timeline}
For consistency, from hereon out, time points 30 minutes and 60 minutes will be termed »early«, while 2 days and 4 days will be referred to as »late«. Differential expression was detected in all groups, lending credibility to the rapid changes in expression needed for a \ac{mir} response of the »immediate-early« type. However, the response to long-term \ac{cntf} stimulation was larger in \ac{mir} numbers as well as effect sizes (Fig. \ref{fig:timepoints-expr-venn} A\&B). Of all early perturbed \acp{mir}, only 3 and 13 \acp{mir} were exclusively perturbed immediate-early-like in \ac{la2} and \ac{la5}, respectively; all others were still \ac{de} after 2 and/or 4 days. In \ac{la2}, the late time points at 2 and, particularly, 4 days showed the greatest perturbation; in \ac{la5}, the picture was more complex (Fig. \ref{fig:timepoints-expr-venn} C\&D). However, generally, there were large similarities as well as exclusivities between the time points 2 and 4 days in both cell lines. When comparing early and late time points between \ac{la2} and \ac{la5} directly, similarly complex patterns emerged (Fig. \ref{fig:timepoints-expr-venn} E\&F). Particularly at late time points (Fig. \ref{fig:timepoints-expr-venn} F), every possible combination of overlap exists. 24\todo{something special?} \acp{mir} were \ac{de} in all late conditions; 107 \acp{mir} were \ac{de} only in \ac{la2}, and 269 \acp{mir} were \ac{de} only in \ac{la5}. \todo{compare most important targets early/late}

\begin{figure} 
\includegraphics[width=\textwidth]{figures/timepoints-expr-venn}
\caption[Timeline Differential Expression.]{\textbf{LA-N-2 / LA-N-5 Timeline and Differential Expression.}
\label{fig:timepoints-expr-venn}}
\end{figure}

\subsubsection{Differential Expression Between LA-N-2 and LA-N-5}
While there was considerable intersection in \ac{de} \acp{mir} between the cell lines, a substantial amount of \acp{mir} was only \ac{de} in one of the two lines. Generally, response to \ac{cntf} was higher in the male-originated \ac{la5} cells; however, there were also \acp{mir} found \ac{de} only in the female \ac{la2} (compare Fig. \ref{fig:timepoints-expr-venn}). Thus, not all of the differences in \ac{mir} expression can be attributed to a higher sensitivity in \ac{la5}. Similarly, \ac{la5} shows a »non-significant trend« toward higher count-change values (mean of absolute count-change across all DE time points, 20,907 versus 3,066, Welch two-sample t test, p = 0.08).

To determine the influence of genotype on the differentiating effect of \ac{cntf}, I employed a statistical interaction design in the DESeq2 Wald test. Briefly, by including an interaction term in the linear regression formula, we can statistically isolate the effect of the condition (\ac{cntf} or control at each time point) between the two genotypes: $$y \sim condition + genotype + condition:genotype$$

Using the interaction term $condition:genotype$, I filtered out the \acp{mir} that reacted significantly different to \ac{cntf} stimulation in \ac{la5} compared to \ac{la2} at each time point. 

\begin{figure} 
\centering
\includegraphics[width=0.5\textwidth]{figures/la2-vs-la5-venn}
\caption[DE miRNAs between LA-N-2 and LA-N-5.]{\textbf{DE miRNAs between LA-N-2 and LA-N-5.}
\label{fig:la2-vs-la5-venn}}
\end{figure}
\begin{figure}
\centering
\includegraphics[width=\textwidth]{figures/la2-vs-la5-overlap-venn}
\caption[LA-N-2 / LA-N-5 DE miRNAs late time points vs in-between.]{\textbf{LA-N-2 / LA-N-5 DE miRNAs late time-points vs in-between.}
\label{fig:la2-vs-la5-overlap-venn}}
\end{figure}
\todo{combine}

intersect with detected miRs in CNTF vs control experiments, 2 venns


To categorise and systematise the sexual dimorphism of \ac{cntf} differentiation of LA-N cells, I performed gene set enrichment of \ac{mir} families in the differential expression datasets.

\subsection{microRNA Family Enrichment}
Of the 151 \ac{mir} families listed in miRBase v21, members of 71 families are \ac{de} in \ac{la2} and \ac{la5}. To test for the enrichment of male, female, and ubiquitously \ac{de} \acp{mir} in these families, I performed gene set enrichment based on Fisher's exact test for each of the families. I found 5 families to be enriched in both male and female cells, and 12 families enriched in only one of the two cell lines (Fig. \ref{fig:mir-de-fam}, left side). The size range of enriched families was substantial, from small families with only 4 mature members to extensive families with dozens of mature \acp{mir}. 

\subsubsection{Gene Targeting of Enriched Families}
Using \textit{miRNet}, the targets of all individual \acp{mir} were determined. Of note, the amount of family members in any \ac{mir} family did not correlate with the absolute amount of targets predicted. Rather, the influence of individual \acp{mir} was the main factor determining the size of the gene target network. However, those families that were enriched in only one cell line presented with significantly smaller target sets than those that were found \ac{de} in both (mean targeted genes per \ac{mir} 217 versus 378, Welch two-sample t test, p = 0.001). Relative to family size, 4 of the enriched families targeted less genes than all others (Fig. \ref{fig:mir-de-fam}, right side): mir-10 (p = 0.016), mir-192 (p = 0.042), mir-379 (p = 0.011), and mir-515 (p < 0.001). Hypothetically, the spectrum of target amounts might correlate with the degree of functional specification of distinct \ac{mir} families: on one end, broadly acting families such as let-7 with sex-independent function, on the other, families with a narrow target profile, such as mir-10, whose restricted function can associate with sex-specific effects.

\begin{figure}
\centering
\includegraphics[width=0.5\textwidth]{figures/mir-de-fam}
\caption[Differential Expression miRNA Family Enrichment]{\textbf{miRNA Families Enriched in Differential Expression.}
\label{fig:mir-de-fam}}
\end{figure}

%\begin{table}
%\centering
%\begin{tabular}{c | c | c | c}
%family name & LA-N-2 p-value & LA-N-5 p-value & implicated in disease\\ \hline
%\hline
%let-7 & 1.2E-3 & 1.5E-7 & disease\\ \hline
%mir & pval & pval & disease\\ \hline
%\end{tabular}
%\caption{Enriched miRNA families in CNTF-induced differential expression in LA-N-2 and LA-N-5.}
%\label{tab:mir-de-fam}
%\end{table}

\section{Network Generation}

\section{The Cholinergic/Neurokine Interface}

\section{Application to Schizophrenia and Bipolar Disorder}
