%%insert blank page?
%\newpage\null\pagestyle{plain}\newpage
%
%% header with section title
%\pagestyle{fancy}
%\fancyhf{}
%\renewcommand{\sectionmark}[1]{\markright{#1}} %no number
%\fancyhead[le,ro]{\nouppercase{\rightmark}}
%\fancyfoot[le,ro]{\thepage}
%\renewcommand{\headrulewidth}{.4pt}
%\renewcommand{\footrulewidth}{.4pt}

%!TEX root = ../dissertation.tex
\chapter{Conclusion}
\label{conclusion}
The study of small RNA dynamics is greatly facilitated by modern bioinformatic methods that enable an understanding of complex transcriptional events in unprecedented detail. While the application and integration of the diverse sources characterising interactions and the participating molecules still are prone to error and misinterpretation, the field advances in giant steps. In summary, the dissertation here presented contributes to the field the following findings:

\begin{itemize}[noitemsep, leftmargin=.5cm, label={\tiny\raisebox{.5ex}{\textbullet}}]
\item Establishment of a framework for the fast and efficient computation of complex interactions between smRNAs, transcription factors, and target genes, including high-resolution tissue specific effects, and on the scale of the whole genome and all miRNAs and tRFs simultaneously;

\item (Re-)establishment of an \emph{in vitro} human cellular model for the study of cholinergic neurons with regard to sex specific phenomena, and, particularly, the effect of neurokine differentiation on cholinergic processes;

\item In this context, elucidation of the potential impact of small RNA dynamics in psychiatric and neurologic diseases;

\item Identification of pertinent mechanisms in the response to stroke of blood-borne human cells, and interactions between small and large transcripts in these cells;

\item Identification of a neuro-immune axis connecting neurokine mechanisms and cholinergic signalling in multiple instances;

\item Exploration of the feasibility and practicality of feedforward loop analyses, in general and in an example of CD14$^+$ monocytes in post-stroke blood samples;

\item Description of a first step in utilising advanced network approaches for data science in the life sciences, for instance by implementing a »smart« dimensionality reduction.
\end{itemize}

In prospect of further research on the subject, this dissertation may be built upon in various ways: \emph{miRNeo} as an infrastructure of smRNA-interactions can be extended by inclusion of novel data on several aspects, such as tissue-specific information on small RNAs (compare Section \ref{sec:stroke:celltypes}) or interactions with small molecule drugs, further enhancing the possibilities of identifying feedforward or other loops relevant to human health. Likewise, integration of data from multiple sources allows the statistical evaluation of data reliability by means of comparison between datasets, increasing the sensitivity and specificity of bioinformatic predictions of relevant interactions. \emph{miRNeo} may also act as an interface between different measurement methods, such as smRNA and mRNA sequencing, or even further parameters such as biomarkers, drug monitoring, or proteome measurements, to enable integrative assessments of complex biological processes in health and disease, and to disentangle the bi-directional relationships between drug effects and smRNA regulation. It may also be extended (incorporating human genomic information) to be used as a de-novo predictive tool for designer oligonucleotides targeting a specific group of transcripts that do not directly mimic endogenous smRNAs, but rather are designed to suppress disease-relevant transcripts with little off-target genomic effects (i.e., toxicogenomics and genocompatibility analyses).

Regardless of the implications of \emph{miRNeo} and related technical issues in computational infrastructure, the analytical properties of network dynamics between small and large RNAs can further enhance our understanding of the principles behind epigenetic regulation in mammalian cells. We are only just beginning to understand the interaction dynamics between small and large RNA in the many-to-many context of molecular reality. More research is needed to comprehend the cell type-specific activation and deactivation of regulatory circuits such as feedforward loops, and their impact on the cell's phenotype and health. Only when the basic principles of smRNA regulation are understood can we begin to rationally build therapeutic strategies that address these mechanisms. 